example/src/main/java/io/netty/example/http2/helloworld/server/Http2ServerInitializer.java

```diff
@@ -90,16 +91,21 @@ private void configureClearText(SocketChannel ch) {
 
         p.addLast(sourceCodec);
         p.addLast(new HttpServerUpgradeHandler(sourceCodec, upgradeCodecFactory));
-        p.addLast(new SimpleChannelInboundHandler<HttpMessage>() {
+        p.addLast(new ChannelInboundHandlerAdapter() {
             @Override
-            protected void channelRead0(ChannelHandlerContext ctx, HttpMessage msg) throws Exception {
-                // If this handler is hit then no upgrade has been attempted and the client is just talking HTTP.
-                System.err.println("Directly talking: " + msg.protocolVersion() + " (no upgrade was attempted)");
-                ChannelPipeline pipeline = ctx.pipeline();
-                ChannelHandlerContext thisCtx = pipeline.context(this);
-                pipeline.addAfter(thisCtx.name(), null, new HelloWorldHttp1Handler("Direct. No Upgrade Attempted."));
-                pipeline.replace(this, null, new HttpObjectAggregator(maxHttpContentLength));
-                ctx.fireChannelRead(msg);
+            public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+                if (msg instanceof HttpMessage) {
+                    HttpMessage httpMsg = (HttpMessage) msg;
+                    // If this handler is hit then no upgrade has been attempted and the client is just talking HTTP.
+                    System.err.println("Directly talking: " + httpMsg.protocolVersion() + " (no upgrade was attempted)");
+                    ChannelPipeline pipeline = ctx.pipeline();
+                    ChannelHandlerContext thisCtx = pipeline.context(this);
+                    pipeline.addAfter(thisCtx.name(), null, new HelloWorldHttp1Handler("Direct. No Upgrade Attempted."));
+                    pipeline.replace(this, null, new HttpObjectAggregator(maxHttpContentLength));
+                    ctx.fireChannelRead(msg);
```

Comment:
@Scottmitch I just wonder if we should for consistency also ensure we release the message once an exception is thrown